### PR TITLE
Domains: Add "Pick Custom Domain" link in domain management for WPCOM domains

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* add "Pick a Custom Domain" in the WPCOM domain vertical navigation

#### Testing instructions

* Open any wpcom domain and check if the "Pick a Custom Domain" vertical navigation item is there
* Verify that it works the same way for .blog subdomains and for wpcomstaging subdomains

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/wpcom-domain.jsx [hunk 1]
```diff
@@ -55,7 +55,43 @@ class WpcomDomain extends React.Component {
...
 	};
 
-	getEditSiteAddressBlock() {
+	handlePickCustomDomainClick = () => {
+		const { domain } = this.props;
+		const domainType = getDomainTypeText( domain );
+
+		this.props.recordGoogleEvent(
+			'Domain Management',
+			`Clicked "Pick a Custom Domain" navigation link on a ${ domainType } in Edit`,
```
Comment (fditrapani) [inline client/my-sites/domains/domain-management/edit/wpcom-domain.jsx]: > Pick a Custom Domain

The upper case Custom Domain isn't necessary. Can we make it all lower case "Pick a custom domain?"

File: client/my-sites/domains/domain-management/edit/wpcom-domain.jsx [hunk 2]
```diff
@@ -55,7 +55,43 @@ class WpcomDomain extends React.Component {
...
+		if ( ! isWpcomDomain ) {
+			return;
+		}
+
+		return (
+			<VerticalNavItem
+				path={ domainAddNew( this.props.selectedSite.slug ) }
+				onClick={ this.handlePickCustomDomainClick }
+			>
+				{ this.props.translate( 'Pick a Custom Domain' ) }
```
Comment (fditrapani) [inline client/my-sites/domains/domain-management/edit/wpcom-domain.jsx]: Same comment about the uppercase C and D

**Comment by klimeryk** [comment]
> I also checked some custom domains and the button did appear there.

It did not appear for me, nor should it appear or custom domains, right?

Other than that, tested as well and all looks good!

**Comment by fditrapani** [comment]
> It did not appear for me, nor should it appear or custom domains, right?

Yeah, sorry. Let's try it here for now.

**Pull request merged**